src/Plugin.php

```diff
@@ -67,6 +67,15 @@ public static function init() {
         remove_action( 'woocommerce_before_shop_loop_item_title', 'woocommerce_template_loop_product_thumbnail', 10);
         add_action( 'woocommerce_before_shop_loop_item_title', __NAMESPACE__ . '\WooCommerce::woocommerce_template_loop_product_thumbnail', 10);
       }
+
+      // Adds video instead of featured image to single product galleries.
+      add_filter('woocommerce_single_product_image_thumbnail_html', __NAMESPACE__ . '\WooCommerce::replaceFeaturedImageWithVideo', 10, 2);
+
+      // Ensures new product are saved before updating its meta data.
+      add_action('woocommerce_process_product_meta', __NAMESPACE__ . '\WooCommerce::saveNewProductBeforeMetaUpdate', 1);
+
+      add_action('woocommerce_process_product_meta', __NAMESPACE__ . '\WooCommerce::woocommerce_process_product_meta');
```

Comment:
```suggestion
      // Saves custom fields for simple products.
      add_action('woocommerce_process_product_meta', __NAMESPACE__ . '\WooCommerce::woocommerce_process_product_meta');
```

Reply:
What about custom fields for other types of products? E.g. Variations?

Reply:
I don't see that we would need videos for each variation just yet. Will add that once useful.